py/client/pydeephaven/query.py

```diff
@@ -163,285 +165,276 @@ def tail(self, num_rows: int):
             table (Table): the right-table of the join
-            on (List[str]): the columns to match, can be a common name or an equal expression,
+            on (Union[str, List[str]]): the column(s) to match, can be a common name or an equal expression,
                 i.e. "col_a = col_b" for different column names
-            joins (List[str], optional): a list of the columns to be added from the right table to the result
-                table, can be renaming expressions, i.e. "new_col = col"; default is empty
+            joins (Union[str, List[str]], optional): the column(s) to be added from the right table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
 
         Returns:
             self
         """
         return super().natural_join(table, on, joins)
 
-    def exact_join(self, table: Any, on: List[str], joins: List[str] = []):
+    def exact_join(self, table: Any, on: Union[str, List[str]], joins: Union[str, List[str]] = None):
         """ Add an exact-join operation to the query.
 
         Args:
             table (Table): the right-table of the join
-            on (List[str]): the columns to match, can be a common name or an equal expression,
+            on (Union[str, List[str]]): the column(s) to match, can be a common name or an equal expression,
                 i.e. "col_a = col_b" for different column names
-            joins (List[str], optional): a list of the columns to be added from the right table to the result
-                table, can be renaming expressions, i.e. "new_col = col"; default is empty
+            joins (Union[str, List[str]], optional): the column(s) to be added from the right table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
 
         Returns:
             self
         """
         return super().exact_join(table, on, joins)
 
-    def join(self, table: Any, on: List[str] = [], joins: List[str] = [], reserve_bits: int = 10):
+    def join(self, table: Any, on: Union[str, List[str]] = None, joins: Union[str, List[str]] = None,
+             reserve_bits: int = 10):
         """ Add a cross-join operation to the query.
 
         Args:
             table (Table): the right-table of the join
-            on (List[str]): the columns to match, can be a common name or an equal expression,
+            on (Union[str, List[str]]): the column(s) to match, can be a common name or an equal expression,
                 i.e. "col_a = col_b" for different column names
-            joins (List[str], optional): a list of the columns to be added from the right table to the result
-                table, can be renaming expressions, i.e. "new_col = col"; default is empty
+            joins (Union[str, List[str]], optional): the column(s) to be added from the right table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
```

Comment:
See the other comments about us not explaining defaults.  There are many more in this file.